File: benchmark/corpus/php/xss_safe_1.php
<?php
// SAFE: all user input passed through htmlspecialchars() before output
$name   = htmlspecialchars($_GET['name'] ?? '', ENT_QUOTES, 'UTF-8');
$search = htmlspecialchars($_POST['q']    ?? '', ENT_QUOTES, 'UTF-8');

echo "<h1>Hello, " . $name . "</h1>";
echo "<p>You searched for: " . $search . "</p>";

File: benchmark/corpus/python/xss_safe_1.py
"""SAFE: user input escaped with markupsafe before embedding in HTML."""
import flask
from markupsafe import escape

app = flask.Flask(__name__)


@app.route("/search")
def search():
    query = escape(flask.request.args.get("q", ""))
    html = f"<html><body><h2>Results for: {query}</h2></body></html>"
    return flask.Response(html, content_type="text/html")
